### PR TITLE
Adds menu item, actions, and reducer for toggling zoom

### LINKED ISSUE #1774: Zoom and pan controls --> OSD wiring, hide / show controls
This adds controls for zooming in, zooming out, and resetting zoom. 

### **Show/hide zoom**
These controls are options at the workspace level, and turn the visible controls on/off for all windows in the workspace. `Show zoom controls` is an option in the workspace menu.

> screen shot 2019-01-29 at 3 01 43 pm

- user selects `Show zoom controls`, 
- the menu closes, 
- the zoom controls are added to the top-right corner of every image canvas in the workspace,
- menu option is replaced by `Hide zoom controls`.

> screen shot 2019-01-25 at 5 39 47 pm

> screen shot 2019-01-29 at 3 03 27 pm

### To close:
- user selects `Hide zoom controls`,
- the menu closes, 
- the zoom controls are added to the top-right corner of every image canvas in the workspace,
- menu option is replaced by `Show zoom controls`.

### **[Zoom independent windows](https://xd.adobe.com/view/722eb97d-f86b-485a-5a03-d4700ae0bc6c-041f/screen/aeee3b37-846d-42da-b4a8-cd097294cdb0/Zoom-controls-on)**

- plus increases the size of the image by [some increment] in the current window only
- minus decreases that image size by the same increment
- revert resets that image to its default size

### Out of scope
- Locked window zoom behavior (controlling zoom levels for multiple windows)
- Visual design for window focus
- Landmark / keyboard controls (OSD comes with some of these built in; larger framework for Mirador's window-focus controls is being discussed in story ticket)

### Resources
- #1660 Zoom and pan story ticket w/ design discussion and accessibility guidance
- [Interactive designs](https://github.com/ProjectMirador/mirador-design/wiki/Mirador-3-Interactive-Designs) --> Zoom controls
- [OpenSeaDragon docs](https://openseadragon.github.io/#examples-and-features)

### PR DESCRIPTION
**Pull request by cbeer**
Co-Authored-By: Shaun Ellis <shaun@sdellis.com>
Co-Authored-By: jkeck <jessie.keck@gmail.com>
Co-Authored-By: cvilla <cvilla100@gmail.com>

Closes #1774

### CODE REVIEW

**Comment by sdellis** [comment]
@cbeer @camillevilla @jkeck  I'm having some trouble with the WorkspaceMenu component test in this PR. When handleZoomToggleClick() is called, I'd like to test that it calls the toggleZoomControls action, but it doesn't seem to be working. First, am I testing the right behavior, and if so does anyone have a tip for why it's not working. 

(I think this is the last test that we need for this PR.)

File: __tests__/src/components/WorkspaceMenu.test.js [hunk 1]
```diff
@@ -5,9 +5,14 @@ import WorkspaceMenu from '../../../src/components/WorkspaceMenu';
 describe('WorkspaceMenu', () => {
   let wrapper;
   let handleClose;
+  let showZoomControls;
+  let toggleZoomControls;
   beforeEach(() => {
     handleClose = jest.fn();
-    wrapper = shallow(<WorkspaceMenu handleClose={handleClose} />);
+    toggleZoomControls = jest.fn();
+    wrapper = shallow(
```
Comment (cbeer) [inline __tests__/src/components/WorkspaceMenu.test.js]:
```suggestion

    wrapper = shallow(
      <WorkspaceMenu
        handleClose={handleClose}
        showZoomControls={showZoomControls}
        toggleZoomControls={toggleZoomControls}
      />,
    );
```

File: __tests__/src/components/WorkspaceMenu.test.js [hunk 2]
```diff
@@ -5,9 +5,14 @@ import WorkspaceMenu from '../../../src/components/WorkspaceMenu';
...
+      <WorkspaceMenu handleClose={handleClose} showZoomControls={showZoomControls} />,
```
Comment (cbeer) [inline __tests__/src/components/WorkspaceMenu.test.js]:
```suggestion
      <WorkspaceMenu
        handleClose={handleClose}
        showZoomControls={showZoomControls}
        toggleZoomControls={toggleZoomControls}
      />,
```

Comment (cbeer) [reply __tests__/src/components/WorkspaceMenu.test.js]: I think you just need to wire up the action through to the element, like ☝️

Comment (sdellis) [reply __tests__/src/components/WorkspaceMenu.test.js]: Yep, that worked and that makes sense. 👍 Thank you!

**Comment by sdellis** [comment]
Ok, looks like there are a few more tests I still need to write for the new zoom control components...

**Comment by sdellis** [comment]
I can't seem to get the ZoomControls to render in a test. I get no errors other than failing tests and empty ShallowWrapper. I followed the testing tips wiki page, but I'm not sure how to debug this.

File: src/components/WorkspaceMenu.js [hunk 3]
```diff
@@ -116,11 +145,15 @@ class WorkspaceMenu extends Component {
...
   handleClose: PropTypes.func.isRequired,
+  toggleZoomControls: PropTypes.func,
+  showZoomControls: PropTypes.bool,
   anchorEl: PropTypes.object, // eslint-disable-line react/forbid-prop-types
 };
 
 WorkspaceMenu.defaultProps = {
   anchorEl: null,
+  showZoomControls: false,
+  toggleZoomControls: () => {},
```
Comment (aeschylus) [inline src/components/WorkspaceMenu.js]: What is the context of using a no-op here?

Comment (aeschylus) [reply src/components/WorkspaceMenu.js]: Ah, nvm.

**Comment by cbeer** [comment]
I don't think there's a strong dependency on #1800, just a similar props shape patten.

**Pull request merged**